lib/rss.js

```diff
@@ -0,0 +1,192 @@
+const RSS = require('rss')
+const generateDiff = require('deep-diff').diff
+const fs = require('fs')
+
+function createLeadersHash(array) {
+  return array.reduce((hash, elem) => {
+    hash[elem.id] = elem
+    return hash
+  }, {})
+}
+
+function createOrgHash(array) {
+  let rank = 0
+  return array.reduce((hash, elem) => {
+    rank++
+    elem.rank = rank
+    elem.leaders = createLeadersHash(elem.leaders)
+    hash[elem.slug] = elem
+    return hash
+  }, {})
+}
+
+let current = {}
+try {
+  current = JSON.parse(
+    fs.readFileSync(`${__dirname}/../out/feed_items.json`).toString()
+  )
+} catch (e) {
+  current = {}
+}
+
+const buildInfo = JSON.parse(
+  fs.readFileSync(`${__dirname}/../out/scrape_info.json`).toString()
+)
+
+const oldData = JSON.parse(
+  fs.readFileSync(`${__dirname}/../out/data_old.json`).toString()
+)
+const newData = JSON.parse(
+  fs.readFileSync(`${__dirname}/../out/data.json`).toString()
+)
+
+const oldOrgs = createOrgHash(oldData)
+const newOrgs = createOrgHash(newData)
+
+console.log(oldOrgs)
+
+if (oldOrgs && newOrgs) {
+  const diffs = generateDiff(oldOrgs, newOrgs) || []
+  console.log(diffs)
+
+  const feedItems = current.items || []
+
+  if (current.lastUpdated !== buildInfo.generated) {
+    diffs.forEach(({ kind, path, lhs, rhs }) => {
+      const organization = oldOrgs[path[0]]
+
+      const itemPath = path.slice(1)
+      const stringPath = itemPath.join('/')
+      const finalProperty = itemPath[itemPath.length - 1]
+
+      let title = ''
+
+      if (itemPath[0] === 'leaders' && itemPath.length === 2 && rhs) {
+        if (kind === 'N') {
+          title = `New Leader for ${organization.name}`
+        } else if (kind === 'D') {
+          title = `Leader Removed from ${organization.name}`
+        }
+
+        return feedItems.push({
+          title,
+          date: newData.lastUpdated,
+          custom_elements: [
+            {
+              'org:name': organization.name,
+            },
+            {
+              'org:slug': organization.slug,
+            },
+            {
+              'org:id': organization.id,
+            },
+            {
+              'property:display_name': rhs.display_name,
+            },
+            {
+              'property:id': rhs.id,
+            },
+            {
+              'property:github_account': rhs.github_account,
+            },
+          ],
+        })
+      }
+
+      if (itemPath[0] === 'leaders') {
+        return
+      }
+
+      if (finalProperty === 'completed_task_instance_count') {
+        title = `Completed Tasks Updated for ${organization.name}`
+      } else if (finalProperty === 'rank') {
+        title = `Organization Rank Updated for ${organization.name}`
+      } else {
+        title = `Updated "${
+          kind === 'A' ? itemPath[itemPath.length - 2] : finalProperty
+        }" for ${organization.name}`
+      }
+
+      if (kind === 'A') {
+        return feedItems.push({
+          title,
```

Comment:
Might want to add description.

The value can be a readable form of `property:old` and `property:new`

Reply:
Done! 👍